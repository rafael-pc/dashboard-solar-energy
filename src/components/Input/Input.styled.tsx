import styled from "styled-components";
import { theme } from "../../styles";

export const InputStyle = styled.input`
  width: 90%;
  height: 38px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 6px;
  margin-top: 5px;
  font-size: ${theme.font.sizes.xxxs};
  border: 1px solid ${theme.colors.gray};
  border-radius: ${theme.border.radius.s};

  &:hover, &:focus-within {
    border: 2px solid ${theme.colors.bg_primary};
  }

  @media only screen and (max-width: ${theme.screen.mobile}) {
    width: 100%;
  }

  @media only screen and (max-height: ${theme.screen.mobile}) {
    font-size: ${theme.font.sizes.xxxxs};
  }
`;

export const Label = styled.div`
  font-family: ${theme.font.family.secondary};
  margin-top: 3px;

  @media only screen and (max-height: ${theme.screen.mobile}) {
    font-size: ${theme.font.sizes.xxxxs};
  }
`;
