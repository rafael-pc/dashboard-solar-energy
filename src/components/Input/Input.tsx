import { FC, InputHTMLAttributes } from "react";
import * as S from "./Input.styled";

interface IInputProps extends InputHTMLAttributes<HTMLInputElement> {
  label?: string;
}

const Input: FC<IInputProps> = ({label, ...props }) => {
  return (
    <>
      <S.Label>{label}</S.Label>
      <S.InputStyle        
        {...props}
      />
    </>
  );
};

export default Input;
