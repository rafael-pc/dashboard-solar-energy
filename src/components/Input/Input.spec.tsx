import { render, screen } from "@testing-library/react";
import Input from "./Input";
import "@testing-library/jest-dom";

describe("Input", () => {
  it("renders the label and input correctly", () => {
    const label = "Test Label";
    const { asFragment } = render(<Input label={label} />);
    expect(asFragment()).toMatchSnapshot();

    const inputLabel = screen.getByText(label);
    expect(inputLabel).toBeInTheDocument();

    const inputElement = screen.getByRole("textbox");
    expect(inputElement).toBeInTheDocument();
  });
});
