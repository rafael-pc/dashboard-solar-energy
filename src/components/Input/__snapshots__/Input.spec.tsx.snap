// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders the label and input correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-grXZZQ ldPpiH"
  >
    Test Label
  </div>
  <input
    class="sc-bcPKhP hWjIqT"
  />
</DocumentFragment>
`;
